src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -126,7 +126,8 @@ public long getLongMetric(String metric)
       return ((Number) metricValue).longValue();
     } else if (metricValue instanceof String) {
       try {
-        return Long.valueOf(((String) metricValue).replace(",", ""));
+        String s = ((String) metricValue).replace(",", "");
+        return s.contains(".") ? Math.round(Double.valueOf(s)) : Long.valueOf(s);
```

Comment:
If the purpose of this PR is to maintain backwards compatibility as your title implies, I think we should maintain the existing behavior.